File: packages/ios/Ariob/Ariob/ViewController.m
//
//  ViewController.m
//  Ariob
//
//  Created by Natnael Teferi on 3/16/25.
//

#import <Lynx/LynxView.h>
#import <Lynx/LynxEnv.h>

#import "ViewController.h"
#import "NetworkLynxProvider.h"

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  LynxView *lynxView = [[LynxView alloc] initWithBuilderBlock:^(LynxViewBuilder *builder) {
    builder.config = [[LynxConfig alloc] initWithProvider:[LynxEnv sharedInstance].config.templateProvider];
    builder.screenSize = self.view.frame.size;
    builder.fontScale = 1.0;
  }];
  lynxView.preferredLayoutWidth = self.view.frame.size.width;
  lynxView.preferredLayoutHeight = self.view.frame.size.height;
  lynxView.layoutWidthMode = LynxViewSizeModeExact;
  lynxView.layoutHeightMode = LynxViewSizeModeExact;
  [self.view addSubview:lynxView];
  
  [lynxView loadTemplateFromURL:@"http://10.0.0.13:3000/main.lynx.bundle?fullscreen=true" initData:nil];
  [lynxView triggerLayout];
}
- (void)viewWillAppear:(BOOL)animated {
  [super viewWillAppear:animated];
  [self.navigationController setNavigationBarHidden:YES animated:NO];
}

@end

File: packages/ios/Ariob/Ariob/ViewController.swift
//
//  File.swift
//  Ariob
//
//  Created by Natnael Teferi on 3/14/25.
//

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()

    let lynxView = LynxView { builder in
      builder.config = LynxConfig(provider: AriobLynxProvider())
      builder.screenSize = self.view.frame.size
      builder.fontScale = 1.0
    }
    
    lynxView.preferredLayoutWidth = self.view.frame.size.width
    lynxView.preferredLayoutHeight = self.view.frame.size.height
    lynxView.layoutWidthMode = .exact
    lynxView.layoutHeightMode = .exact
    self.view.addSubview(lynxView)

    lynxView.loadTemplate(fromURL: "main.lynx", initData: nil)
  }
}
